### PR TITLE
New watch order methods

### PR DESCRIPTION
**Pull request by matanporeh**
asana https://app.asana.com/0/32014397880520/298717100238135

### CODE REVIEW

File: sources/BringgTrackingClient.h [hunk 1]
```diff
@@ -86,15 +86,28 @@
...
  *
  *  @param uuid       order uuid
  *  @param shareduuid shared uuid
  *  @param delegate   delegate
- *  @throws if invalid or missing order UUID
+ *  @throws if invalid or missing either order UUID or shared UUID
  */
 - (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
-                        sharedUUID:(NSString *_Nullable)shareduuid
+                        sharedUUID:(NSString *_Nonnull)shareduuid
```
Comment (orxelm) [inline sources/BringgTrackingClient.h]: should be shareUUID

File: sources/BringgTrackingClient.h [hunk 2]
```diff
@@ -107,9 +120,20 @@
  *  @see DriverDelegate
  */
 - (void)startWatchingDriverWithUUID:(NSString *_Nonnull)uuid
```
Comment (orxelm) [inline sources/BringgTrackingClient.h]: should be shareUUID

File: sources/BringgTrackingClient.h [hunk 3]
```diff
@@ -207,17 +230,36 @@ completionHandler:(nullable GGRatingResponseHandler)completionHandler;
...
+- (BOOL)isWatchingDriverWithUUID:(NSString *_Nonnull)uuid;
 
 
 /**
+ *  return a shared uuid object matching a driver uuid
+ *
+ *  @param uuid driver uuid to search by
+ *
+ *  @return NSString
+ */
```
Comment (orxelm) [inline sources/BringgTrackingClient.h]: should be
```objc
- (nullable NSString *)shareUUIDForDriverWithUUID:(nonnull NSString*)driverUUID;
```

File: sources/BringgTrackingClient.m [hunk 4]
```diff
@@ -149,22 +149,89 @@ - (nullable GGCustomer *)signedInCustomer{
 
 
 - (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
-                        sharedUUID:(NSString *_Nullable)shareduuid
```
Comment (orxelm) [inline sources/BringgTrackingClient.m]: Again, change all shareduuid to shareUUID or sharedUUID (will stop commenting regarding this)

File: sources/BringgTrackingClient.m [hunk 5]
```diff
@@ -149,22 +149,89 @@ - (nullable GGCustomer *)signedInCustomer{
...
+    }
     
-    [self.trackerManager startWatchingOrderWithUUID:uuid sharedUUID:shareduuid delegate:delegate];
+    [self.trackerManager startWatchingOrderWithUUID:uuid accessControlParamKey:PARAM_ACCESS_TOKEN accessControlParamValue:customerAccessToken delegate:delegate];
 }
 
 
 
 - (void)startWatchingDriverWithUUID:(NSString *_Nonnull)uuid
-                          shareUUID:(NSString *_Nonnull)shareUUID
```
Comment (orxelm) [inline sources/BringgTrackingClient.m]: I see that you sometimes write shareUUID and sometimes share**d**UUID so be consistent (with android also)

File: sources/model/managars/GGTrackerManager.m [hunk 6]
```diff
@@ -715,12 +704,10 @@ - (void)notifyRESTFindMeUpdatedForOrderWithUUID:(NSString * _Nonnull)orderUUID{
...
-     GGDriver *driver = [self.liveMonitor getDriverWithUUID:driverUUID];
-    
-     NSString *compoundKey = [[driverUUID stringByAppendingString:DRIVER_COMPOUND_SEPERATOR] stringByAppendingString:shareUUID];
+     GGDriver *driver = [self driverWithUUID:driverUUID];
     
     // update the order delegate
-    id<DriverDelegate> delegate = [_liveMonitor.driverDelegates objectForKey:compoundKey];
+    id<DriverDelegate> delegate = [_liveMonitor.driverDelegates objectForKey:driverUUID];
     
     if (delegate) {
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: respondsToSelector....

File: tests/BringgTrackingClientTests.m [hunk 7]
```diff
@@ -0,0 +1,550 @@
+//
+//  BringgTrackingTests.m
```
Comment (orxelm) [inline tests/BringgTrackingClientTests.m]: What is the coverage of this testclass?

File: sources/BringgTrackingClient.h (hunk 1 above)
Comment (matanporeh) [inline sources/BringgTrackingClient.h]: done

File: sources/BringgTrackingClient.h (hunk 2 above)
Comment (matanporeh) [inline sources/BringgTrackingClient.h]: done

File: sources/BringgTrackingClient.h (hunk 3 above)
Comment (matanporeh) [inline sources/BringgTrackingClient.h]: ok

File: sources/BringgTrackingClient.m (hunk 4 above)
Comment (matanporeh) [inline sources/BringgTrackingClient.m]: ok.

File: sources/BringgTrackingClient.m (hunk 5 above)
Comment (matanporeh) [inline sources/BringgTrackingClient.m]: replaced

File: sources/model/managars/GGTrackerManager.m (hunk 6 above)
Comment (matanporeh) [inline sources/model/managars/GGTrackerManager.m]: fixed

File: tests/BringgTrackingClientTests.m (hunk 7 above)
Comment (matanporeh) [inline tests/BringgTrackingClientTests.m]: for the whole class its about 45%
for the watch methods its almost 100% (start watch waypoint has no tests)
if you wish to increase the coverage for methods that are outside the scope of this PR - create an asana for it
